appscale/tools/appscale_stats.py

```diff
@@ -0,0 +1,626 @@
+  'process.cpu': ['percent'],
+  'process.children_stats_sum': ['memory', 'cpu']
+}
+
+# Needed node field list to processes statistics
+INCLUDE_NODE_LIST_FOR_PROCESSES = {
+  'node': ['cpu'],
+  'node.cpu': ['count']
+}
+
+# Needed field list to proxies statistics
+INCLUDE_PROXY_LIST = {
+  'proxy': ['unified_service_name', 'application_id', 'servers_count',
+            'frontend', 'backend'],
+  'proxy.frontend': ['req_rate', 'req_tot', 'hrsp_5xx', 'hrsp_4xx',
+                     'bin', 'bout', 'scur'],
+  'proxy.backend': ['qtime', 'rtime', 'qcur']
+}
+
+
+def _get_stats(keyname, path, include_lists=None):
+  """
+  Returns statistics from Hermes.
+
+  Args:
+    keyname: A string represents an identifier from AppScaleFile.
+    path: A string represents one of 'nodes', 'processes' or 'proxies'.
+    include_lists: A list represents a list of desired fields.
+  """
+  if not include_lists:
+    include_lists = {}
+    if path is 'nodes':
+      include_lists.update(INCLUDE_NODE_LIST)
+    if path is 'processes':
+      include_lists.update(INCLUDE_PROCESS_LIST)
+    if path is 'proxies':
+      include_lists.update(INCLUDE_PROXY_LIST)
+
+  login_host = LocalState.get_login_host(keyname=keyname)
+  secret = LocalState.get_secret_key(keyname=keyname)
+  hermes_port = "17441"
+  stats_path = "/stats/cluster/{path}".format(path=path)
+
+  headers = {
+    'Appscale-Secret': str(secret)
+  }
+
+  data = {
+    'include_lists': include_lists
+  }
+
+  url = "http://{ip}:{port}{path}".format(
+    ip=login_host,
+    port=hermes_port,
+    path=stats_path
+  )
+
+  resp = requests.get(url=url, headers=headers, data=json.dumps(data))
+
+  return resp.json()["stats"]
+
+
+def show_stats(options):
+  """
+  Prints nodes, processes and/or proxies statistics nicely.
+
+  Args:
+    options: A Namespace that has fields for each parameter that can be
+      passed in via the command-line interface.
+
+  Raises:
+    AttributeError: if some of statistic is equal None.
+  """
+  try:
+    nodes_stats = None
+    processes_stats = None
+    proxies_stats = None
+
+    if options.show:
+      if "nodes" in options.show:
+        nodes_stats = []
+      if "processes" in options.show:
+        processes_stats = []
+      if "proxies" in options.show:
+        proxies_stats = []
+
+    verbose = False
+    if options.verbose:
+      verbose = True
+
+    if nodes_stats is not None:
+      nodes_stats = get_node_stats(keyname=options.keyname,
+                                   specified_roles=options.roles,
+                                   verbose=verbose)
+
+    if processes_stats is not None:
+      summary = True
+      top = None
+      if options.verbose:
+        summary = False
+      if "name" in options.order_processes:
+        sort = 0 if summary else 1
+      elif "mem" in options.order_processes:
+        sort = 2 if summary else 2
+      elif "cpu" in options.order_processes:
+        sort = 3 if summary else 3
```

Comment:
In case when you set "choices" property in argparse library it returns a list but not a string.